amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/dashboard/controller/OptimizerController.java

```diff
@@ -64,8 +66,25 @@ public void getOptimizerTables(Context ctx) {
     Integer pageSize = ctx.queryParamAsClass("pageSize", Integer.class).getOrDefault(20);
     int offset = (page - 1) * pageSize;
 
+    // get all info from underlying table table_runtime
+    List<TableRuntimeBean> tableRuntimeBeans;
+    if (ALL_GROUP.equals(optimizerGroup)) {
+      tableRuntimeBeans = tableService.getTableRuntimesForAllGroup(pageSize, offset);
+    } else {
+      tableRuntimeBeans = tableService.getTableRuntimes(optimizerGroup, pageSize, offset);
+    }
+
     List<TableRuntime> tableRuntimes = new ArrayList<>();
-    List<ServerTableIdentifier> tables = tableService.listManagedTables();
+    List<ServerTableIdentifier> tables = new ArrayList<>();
+    for (TableRuntimeBean bean : tableRuntimeBeans) {
```

Comment:
It seems that you only use TableRuntimeBean to build TableIdentifier.
I think you could directly use TableRuntimeBean instead of TableRuntime